### PR TITLE
chore(sdk): add test and docs for batch usage of ulid

### PR DESCRIPTION
**Pull request by skorfmann**
this adds docs (and a test) for the fact that ulid works with a 1ms precision only, which might cause inconsistent ordered ids for multiple calls to `ulid` at the exact same time. The generated ids are still unique, but might not reflect the order of creation.

## Checklist

- [ ] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [ ] Description explains motivation and solution
- [ ] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: examples/jsii-fixture/dist/js/jsii-fixture@0.0.0.jsii.tgz
Comment (Chriscbr) [inline examples/jsii-fixture/dist/js/jsii-fixture@0.0.0.jsii.tgz:1]: Remove

File: docs/docs/04-standard-library/util/api-reference.md [hunk 1]
```diff
@@ -256,6 +256,13 @@ util.ulid(options?: UlidOptions);
 
 Generates universally unique lexicographically sortable identifier.
 
+This works with a precision of 1ms, which might not be sufficient for some cases where the strict order of
+creation is a desired property - e.g. batch creation of resources. In these cases, the generated IDs will
+be still unique (can generate 1.21e+24 unique ULIDs per millisecond) but might not be strictly reflecting the order of creation.
+To mitigate this, you could sleep for 1ms between creation of resources. However, in distributed
+systems where the strict order of creation is a requirement, this is not sufficient
+and you should consider using a different ID generation strategy.
```
Comment (Chriscbr) [inline docs/docs/04-standard-library/util/api-reference.md:264]: Out of curiosity are you still using `ulid` for your use case or did you switch to another library/solution? (just trying to gauge usage)

File: examples/tests/valid/ulid.test.w [hunk 2]
```diff
@@ -0,0 +1,35 @@
+bring util;
+bring expect;
+
+inflight class Sortable {
+  pub static extern "./sort.js" sortStringListAsc(arr: Array<str>): Array<str>;
+  pub static extern "./sort.js" sortStringListDesc(arr: Array<str>): Array<str>;
```
Comment (Chriscbr) [inline examples/tests/valid/ulid.test.w:6]: Maybe we can add a note to remove these in the future
```suggestion
  // TODO: https://github.com/winglang/wing/issues/4998
  pub static extern "./sort.js" sortStringListAsc(arr: Array<str>): Array<str>;
  pub static extern "./sort.js" sortStringListDesc(arr: Array<str>): Array<str>;
```

**Review by Chriscbr — APPROVED** [review]

File: examples/tests/valid/ulid.test.w [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+}
+
```
Comment (staycoolcall911) [inline examples/tests/valid/ulid.test.w:8]: Makes sense to test it in preflight as well, right?

**Review by staycoolcall911 — APPROVED** [review]

File: docs/docs/04-standard-library/util/api-reference.md (hunk 1 above)
Comment (skorfmann) [reply docs/docs/04-standard-library/util/api-reference.md:264]: not using it right now, but would personally use this for most of the cases where I want to generate ids for records.

File: examples/tests/valid/ulid.test.w (hunk 3 above)
Comment (skorfmann) [reply examples/tests/valid/ulid.test.w:9]: just tried it, but ran into issues with other tooling to force a sleep. so just added a simple preflight test.

- https://github.com/winglang/wing/pull/5417
- https://github.com/winglang/wing/issues/5418

**Pull request closed without merging**